tests/nested/manual/core-early-config/task.yaml

```diff
@@ -0,0 +1,74 @@
+summary: Test that config defaults are applied early when image is created.
+
+# core18 specific test (and nested vm is derived from host system)
+systems: [ubuntu-18.04-64]
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    cleanup_nested_env
+
+    # nested suite installs latest snapd only in the nested system; we need
+    # new snapd on the host as well for prepare-image changes.
+    dpkg -i "$SPREAD_PATH"/../snapd_*.deb
+
+    mkdir extra-snaps
+
+    # modify and repack gadget snap (add defaults section and install hook)
+    snap download --channel=18/stable pc
+    GADGET_SNAP=$(ls pc_*.snap)
+    unsquashfs -no-progress "$GADGET_SNAP"
+    rm -f "$GADGET_SNAP"
+    cat defaults.yaml >> squashfs-root/meta/gadget.yaml
+    mkdir -p squashfs-root/meta/hooks
+    cp install squashfs-root/meta/hooks/
+    mksquashfs squashfs-root "$GADGET_SNAP" -comp xz -no-fragments -no-progress
+    rm -rf squashfs-root
+    mv "$GADGET_SNAP" extra-snaps/
+
+    # XXX: pc-kernel initrd fix; this will be removed once initrd changes land
+    snap download --channel=18/edge pc-kernel
+    KERNEL_SNAP=$(ls pc-kernel_*.snap)
+    mv "$KERNEL_SNAP" extra-snaps/
+
+    create_nested_core_vm
+
+    # Modify seed to use devmode for pc gadget snap. This is needed for the
+    # install hook to have access to /etc/systemd. Ideally we would use
+    # system-files plug, but it wouldn't get autoconnected due to assertions. 
+    IMAGE_PATH="$(get_nested_core_image_path)"
+    loops=$(kpartx -avs "$IMAGE_PATH"  | cut -d' ' -f 3)
+    part=$(echo "$loops" | tail -1)
+    tmp=$(mktemp -d)
+    mount "/dev/mapper/$part" "$tmp"
+    sed -i "$tmp/system-data/var/lib/snapd/seed/seed.yaml" -E -e 's/^(\s+)file: pc_x1.snap/\1file: pc_x1.snap\n\1devmode: true\n/'
+    umount "$tmp"
+    kpartx -d "$IMAGE_PATH"
+    rmdir "$tmp"
+
+    start_nested_core_vm
+
+kill-timeout: 40m
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    destroy_nested_vm
+    cleanup_nested_env
+    rm -rf extra-snaps
+
+debug: |
```

Comment:
Debug section is not running any command to display information

Reply:
Thanks for spotting, removed.